metaflow/task.py

```diff
@@ -513,19 +514,33 @@ def run_step(
                 }
             }
         )
-        logger = self.event_logger
+
+        # 6. Update event logger with additional context around run_id, step_name, task_id, retry_count, and
+        # project_info to event logger context
+        # We don't update the context for the monitor as the cardinality of this context is high
+        event_logger_context = {
+            "run_id": run_id,
+            "step_name": step_name,
+            "task_id": task_id,
+            "retry_count": retry_count,
+            "project_name": current.get("project_name"),
+            "branch_name": current.get("branch_name"),
+            "is_user_branch": current.get("is_user_branch"),
+            "is_production": current.get("is_production"),
+            "project_flow_name": current.get("project_flow_name"),
+            "trace_id": trace_id or None,
+        }
+        self.event_logger.send(
```

Comment:
nit: could we use `system_current` to be consistent? It feels like we could get rid of the monitor/logger passed in to the decorators/task (it's a good thing) and use the globally present one.